lib/omniauth/strategies/oauth2-multiple-state.rb

```diff
@@ -73,19 +73,19 @@ def token_params
 
       def callback_phase # rubocop:disable AbcSize, CyclomaticComplexity, MethodLength, PerceivedComplexity
         error = request.params["error_reason"] || request.params["error"]
-        if error
-          fail!(error, CallbackError.new(request.params["error"], request.params["error_description"] || request.params["error_reason"], request.params["error_uri"]))
-        elsif !options.provider_ignores_state && (request.params["state"].to_s.empty? || session["omniauth.states"].to_s.empty? || !session["omniauth.states"].delete(request.params["state"]))
-          fail!(:csrf_detected, CallbackError.new(:csrf_detected, "CSRF detected"))
-        else
-          self.access_token = build_access_token
-          self.access_token = access_token.refresh! if access_token.expired?
-
-          if session['omniauth.state_origins'] && session['omniauth.state_origins'][request.params['state']]
-            env['omniauth.origin'] = session['omniauth.state_origins'].delete(request.params['state'])
-          end
-          super
+
+        fail_with_error(:session_expired_on_tab, { "error" => "session_expired_on_tab", "error_description" => "Session expired on tab" }) if session_expired_on_tab?(error, request.params["error_description"])
+        fail_with_error(error, request.params) if error
+        fail_with_error(:csrf_detected, { "error" => "csrf_detected", "error_description" => "CSRF detected" }) if csrf_detected?
```

Comment:
```suggestion
        fail_with_error(:csrf_detected, { "error" => "csrf_detected", "error_description" => "CSRF detected" }) if csrf_detected?(request.params)
```

Akışı bi deneyelim başka gözden kaçan bir şey olmasın